### PR TITLE
Allowlisting remote URLs for displaying content on the content dashboard

### PR DESCRIPTION
**Pull request by elit0451**
## Details
- Removed a non-existent setting from the Web.config (`ContentDashboardBaseUrl`), left out after the renaming done in commit c6bf78e6728caf15afa186597ee105d6de121937;
- Added the ability to allowlist the remote URLs for the "Getting Started" dashboard in config.
- If no addresses are specified, any address is allowed.

## Test
- Without specifying any addresses, the "Getting Started" dashboard should display the content as before.
- Add the following line in Web.config and verify that the behaviour persists:
```
    <appSettings>
        ...
        <add key="Umbraco.Core.ContentDashboardUrl-Allowlist" value="https://dashboard.umbraco.org/,https://dashboard.umbraco.com/,https://www.google.com/"/>
        ...
    </appSettings>
```
- Remove either https://dashboard.umbraco.com/ (and observe that the content is not fetched) or https://dashboard.umbraco.org/ (and observe that the CSS is not fetched) 
  - Check that the corresponding calls to the **GetRemoteDashboardContent** or **GetRemoteDashboardCss** endpoints returns 400 and that the errors are also in the log (Settings > Log Viewer)

### CODE REVIEW

**Review by nikolajlauridsen — COMMENTED** [review]
Looks good overall, only found one minor thing 👍

File: src/Umbraco.Web/Editors/DashboardController.cs [hunk 1]
```diff
@@ -61,6 +61,13 @@ public async Task<JObject> GetRemoteDashboardContent(string section, string base
             var version = UmbracoVersion.SemanticVersion.ToSemanticString();
             var isAdmin = user.IsAdmin();
 
+            if(!IsAllowedUrl(baseUrl))
+            {
+                var errorMsg = $"The following URL is not listed in the allowlist for ContentDashboardUrl in the Web.config: {baseUrl}";
+                Logger.Error<DashboardController>(errorMsg);
+                throw new HttpResponseException(Request.CreateErrorResponse(HttpStatusCode.BadRequest, errorMsg));
```
Comment (nikolajlauridsen) [inline src/Umbraco.Web/Editors/DashboardController.cs:68]: I don't think we should include the full error message in the error response, only in the log entry, and instead just return something more generic like "Could not load dasboard"

Comment (elit0451) [reply src/Umbraco.Web/Editors/DashboardController.cs]: Makes sense 🙌 Will fix up

**Review by nikolajlauridsen — APPROVED** [review]
LGTM now, will merge when tests are happy :)

**Pull request merged**